### ISSUE TITLE
Fatal error: Class 'Mage' not found

### ISSUE THREAD

**Issue by clockworkgeek** [magento/magento2#167]
I have "display_errors" set to "on" for my dev server.  Sometimes, in backend, an error is shown at the bottom of a page after all other output.  I have the latest dev-34 version.

```
Fatal error: Class 'Mage' not found in /var/www/app/code/core/Mage/Core/Model/Abstract.php on line 174

Call Stack:
    0.9556   17248592   1. Mage_Core_Model_Abstract->__sleep() /var/www/app/code/core/Mage/Core/Model/Abstract.php:0
```

Clearly a model is being serialized as it is disposed automatically.  But the 'Mage' class has already been disposed before it.

**Comment by magento-engcom-team** [comment]
Hi @clockworkgeek. Thank you for your report. 
The issue has been fixed in magento/magento2#21731 by @VitaliyBoyko in 2.3-develop branch
Related commit(s):
 - [3819dc1dcc2aa19c3811e33b88c8f31f599f8428](https://github.com/magento/magento2/commit/3819dc1dcc2aa19c3811e33b88c8f31f599f8428)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @clockworkgeek. Thank you for your report. 
The issue has been fixed in magento/magento2#21793 by @VitaliyBoyko in 2.2-develop branch
Related commit(s):
 - [27d113fc9295cff8909913a41867d64dbbcfb377](https://github.com/magento/magento2/commit/27d113fc9295cff8909913a41867d64dbbcfb377)

The fix will be available with the upcoming 2.2.9 release.
